lib/scarlet/server.rb

```diff
@@ -46,8 +48,10 @@ def initialize cfg
       @current_nick = config.nick
       config.control_char ||= Scarlet.config.control_char
       config.buffer_rate ||= 1
+      config.scheduler_rate ||= 1
       config.freeze
 
+      @reminder_scheduler = ReminderScheduler.new self
```

Comment:
Why don't you pass the rufus scheduler into the task scheduler and use it for the actual job running instead of reimplementing all of that logic? Then all you'd need to do is restore any of the saved tasks upon process boot